etna/records/blocks.py

```diff
@@ -55,13 +55,15 @@ def get_prep_value(self, value):
         return value.iaid
 
     def bulk_to_python(self, values):
-        """Return the model instances for the given list of primary keys.
+        """Return a list of model instances for the given list of primary keys.
         The instances must be returned in the same order as the values and keep None values.
         """
-        _, records = self.target_model.api.fetch_all(iaids=values)
-        record_dict = {r.iaid: r for r in records}
-        record_list = tuple(record_dict.get(iaid) for iaid in values)
-        return record_list
+        if not values:
```

Comment:
The question is why does the function gets called if the values not true?

Reply:
I would guess it has something to do with the state of the page... maybe there are instances of the block that have lost their values due to the other bug we fixed.